File: challenger-backend/src/main/java/com/kameo/challenger/odb/TaskApprovalODB.java
package com.kameo.challenger.odb;

import com.kameo.challenger.odb.api.IIdentity;
import com.kameo.challenger.web.rest.api.IChallengerService;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.crsh.cli.Man;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import static com.kameo.challenger.odb.TaskApprovalODB_.rejectionReason;

@Data
@NoArgsConstructor
@ToString(of = "id")
@Entity
public class TaskApprovalODB implements IIdentity {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @NotNull
    @ManyToOne
    TaskODB task;

    @NotNull
    @ManyToOne
    UserODB user;

    @NotNull
    @Enumerated
    TaskStatus taskStatus;

    String rejectionReason;


    @Override
    public boolean isNew() {
        return getId()<0;
    }


}

File: challenger-backend/src/main/java/com/kameo/challenger/odb/TaskApprovalODB.kt
package com.kameo.challenger.odb

import com.kameo.challenger.odb.api.IIdentity
import javax.persistence.*


@Entity
data class TaskApprovalODB(@Id
                           @GeneratedValue(strategy = javax.persistence.GenerationType.AUTO)
                           override var id: Long = 0) : IIdentity {

    @ManyToOne
    lateinit var task: TaskODB;

    @ManyToOne
    lateinit var user: UserODB;

    @Enumerated
    lateinit var taskStatus: TaskStatus;

    var rejectionReason: String? = null
}